docs/user_guides/tls_proxy.md

```diff
@@ -419,16 +416,61 @@ openssl x509 -req -days 3650 -in client.csr -out client.pem -CA ca.pem -CAkey ca
 ```
 </details>
 
+The ```openssl.cnf``` file is a default configuration file of the openssl. If you want to know more details, visit [the manual page](https://www.openssl.org/docs/manmaster/man5/config.html). The ```gen.sh``` file can generate some related certificate files. And you can modify this file to customize for specific demands, such as the valid period of the certificate file, the length of the certificate key or the certificate file names.
+
+In the ```gen.sh``` file, these variables are very important, because they are required in the process of creating a certificate signing request file. The first five variables are the basic signing information, including country, state, location, organization, organizational. CommonName is a useful configuration that deserves your attention. Because this configuration will be verified when the client and the server communicate with each other
+
 ### 3. Run gen.sh to create certificate.
 ```shell
 chmod +x gen.sh
 ./gen.sh
 ```
-They will creat 9 files about ca.key, ca.pem, ca.srl, server.key, server.pem, server.csr, client.key, client.pem, client.csr.
 
-## Modify config
+Finally, you will get 9 files, including ca.key, ca.pem, ca.srl, server.key, server.pem, server.csr, client.key, client.pem, client.csr. If you want to know the ```gen.sh``` file detail, read the next chapter.
+
+### 4. Get the detail of generating certificate files
+
+To implement the ssl/tls mutual authentication, it mainly involves three parts, including ca, client, server. The ca is to ensure that the certificate used by the client and server to communicate with each other is legal.
+
+> Tips: If you want to know more details of using the openssl command, execute ```man openssl``` or visit [the openssl manual page](https://www.openssl.org/docs/)
+
+- Generate an RSA private key for the ca. If you want to know more details of this command, execute ```openssl genras -help``` or ```man openssl```.
+
+```shell
+openssl genpkey -algorithm RSA -out ca.key
+```
+
+- Make a certificate request and generate a ca certificate. This process needs some basic information of ca and you can get a ```ca.pem``` file which is a ca certificate used to generate client and server certificates. The ```x509``` option outputs a certificate instead of a certificate request, which means self-signing.
+
+```shell
+openssl req -new -x509 -key ca.key -out ca.pem -days 3650 -subj "/C=$Country/ST=$State/L=$Location/O=$Organization/OU=$Organizational/CN=$CommonName"
+```
+
+Generating a server certificate is same as generating a client certificate, including three steps. The following is an example of a server to illustrate this process.
+
+- Generate a server private key, you will get a ```server.key``` file.
+
+```shell
+openssl genpkey -algorithm RSA -out server.key
+```
+
+- Generate a certificate signing request file, and some basic server info is required. When the command complete to execute, you will get a ```server.csr``` file.
```

Comment:
Generate a certificate signing request file, and provide the required information about the server. After running the following command, you will get a `server.csr` file.